var/spack/repos/builtin/packages/ascent/package.py

```diff
@@ -140,9 +142,10 @@ def install(self, spec, prefix):
             cmake(*cmake_args)
             make()
             make("install")
-            # TODO also copy host_cfg_fname into install
+            # install copy of host config for provenance
+            install(host_cfg_fname, prefix)
 
-    def create_host_config(self, spec, prefix):
+    def create_host_config(self, spec, prefix, py_site_pkgs_dir=None):
```

Comment:
Could you add a comment which refers to https://github.com/spack/spack/issues/6261 and/or explain that this ensures that the site-packages directory is defined when this is called from a subclass?

As a side note, when you mention:

> It's not clear the base site-packages dir would be correct

I'm not sure what you mean. The spec you would be working with here is the subclass spec, so you could construct the subclass-specific site packages directory. That being said the way it's done here seems straightforward (if the comment is added).

Reply:
sure I can add a comment about #6261 

The "it's not clear" is b/c the `site_package_dir` is set as a module var, not as part of the package instance.  But it seems to be set to specific install location of the package instance. 

Even finding the right path, what would the base class `site_packages_dir` be set to?
The right answer is not clear to me. The mix of inheritance + magic makes it a bit hard for me the comprehend. But thats all fine, I think the solution I have is good b/c it allows me to still use spack's site_packages_dir as intended for the main packages.